core/src/main/java/org/hisp/dhis/android/core/datavalue/internal/DataValuePostCall.java

```diff
@@ -74,6 +80,8 @@
                 return systemInfoDownloader.downloadMetadata().andThen(Observable.create(emitter -> {
                     emitter.onNext(progressManager.increaseProgress(SystemInfo.class, false));
 
+                    markPartitionsAsUploading(dataValues);
+
                     DataValueSet dataValueSet = new DataValueSet(dataValues);
 
                     DataValueImportSummary dataValueImportSummary = apiCallExecutor.executeObjectCall(
```

Comment:
Wrap this part in a try-catch to mark the partition as error in case the response response is an error. Otherwise the partition will remain in UPLOADING state